### ISSUE TITLE
Hardware back does not close app

### ISSUE THREAD

**Issue by SMarijic** [shprink/ionic-native-transitions#47]
When there is no back view, hardware back button should close app, but instead it does nothing. I think problem is this piece of code

if (!$ionicHistory.backView()) {
    return;
}

I tried to put  ionic.Platform.exitApp(); instead of return, it does close the app but it's still in memory, so when I try to open it back it causes "White Screen of Death" but that is probably whole different problem.

**Comment by shprink** [comment]
It does close the app for me. what version of this plugin do you use? what version of telerik?

**Comment by SMarijic** [comment]
I'm using rc-5 version of plugin, and 0.5.4 of telerik

**Comment by SMarijic** [comment]
@shprink On which version of Android you did test?

**Comment by shprink** [comment]
Android 6

**Comment by arielfaur** [comment]
I see the same behavior in my app, back button does nothing. I am on Android 6 with rc6 and Telerik 0.5.5

**Comment by shprink** [comment]
Does nothing all the time or just when no history?

Do you use the ion-back-button directive ?

**Comment by arielfaur** [comment]
It works fine in all scenarios, except when there is no history (e.g. the first view in the stack). 
Yes, I have the default setup in the index.html:
````
<ion-nav-bar class="bar-positive">
      <ion-nav-back-button>
      </ion-nav-back-button>
</ion-nav-bar>

<ion-nav-view></ion-nav-view>
````

**Comment by shprink** [comment]
ok you are not the first one to report that so i am going to investigate.
Can you send your app in debug mode to me?

**Comment by arielfaur** [comment]
Unfortunately I cannot send the app because I signed a confidential agreement. But I can try to replicate the issue in a new app if you want. I just need to find the time for it :)

**Comment by shprink** [comment]
that would be great because I cannot reproduce in my end :(

**Comment by FreezeSoul** [comment]
$ionicPlatform.registerBackButtonAction(function () {
                        if(!$ionicHistory.backView()){
                            if(navigator.app){
                                navigator.app.exitApp();
                            }
                        }else{
                            if($rootScope.$ionicGoBack){
                                $rootScope.$ionicGoBack();
                            }
                        }
                    }, 101);

**Comment by shprink** [comment]
Fixed in v1.0.0-rc9
